src/ngAnimate/animateCss.js

```diff
@@ -623,6 +623,7 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
 
       if (options.delay != null) {
         var delayStyle = parseFloat(options.delay);
+        maxDelay = Math.max(options.delay, 0);
```

Comment:
Hmm. The maxDelay is mainly used for determining the closing timeout. And options.delay overwrites values set in the transition styles.